aws/PAS-network-environment-NAT.json

```diff
@@ -1,1655 +1,2102 @@
+      "Type": "AWS::EC2::Route",
+      "Properties": {
+        "DestinationCidrBlock": "0.0.0.0/0",
+        "RouteTableId": {
+          "Ref": "PASPublicRT"
+        },
+        "GatewayId": {
+          "Ref": "PASIGW"
+        }
+      },
+      "DependsOn": "PASGWAttachment"
+    },
+    "VaultSGIngress1": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {
+          "Ref": "VaultSG"
+        },
+        "IpProtocol": "tcp",
+        "FromPort": 1858,
+        "ToPort": 1858,
+        "SourceSecurityGroupId": {
+          "Ref": "VaultSG"
+        },
+        "Description": "Vault to Vault connection"
+      }
+    },
+    "VaultSGIngress2": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {
+          "Ref": "VaultSG"
+        },
+        "IpProtocol": "tcp",
+        "FromPort": 1858,
+        "ToPort": 1858,
+        "SourceSecurityGroupId": {
+          "Ref": "PVWASG"
+        },
+        "Description": "Vault to PVWA connection"
+      }
+    },
+    "VaultSGIngress3": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {
+          "Ref": "VaultSG"
+        },
+        "IpProtocol": "tcp",
+        "FromPort": 1858,
+        "ToPort": 1858,
+        "SourceSecurityGroupId": {
+          "Ref": "CPMSG"
+        },
+        "Description": "Vault to CPM connection"
+      }
+    },
+    "VaultSGIngress4": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {
+          "Ref": "VaultSG"
+        },
+        "IpProtocol": "tcp",
+        "FromPort": 1858,
+        "ToPort": 1858,
+        "SourceSecurityGroupId": {
+          "Ref": "PTASG"
+        },
+        "Description": "Vault to PTA connection"
+      }
+    },
+    "VaultSGIngress5": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {
+          "Ref": "VaultSG"
+        },
+        "IpProtocol": "tcp",
+        "FromPort": 1858,
+        "ToPort": 1858,
+        "SourceSecurityGroupId": {
+          "Ref": "PSMSG"
+        },
+        "Description": "Vault to PSM connection"
+      }
+    },
+    "VaultSGIngress6": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {
+          "Ref": "VaultSG"
+        },
+        "IpProtocol": "tcp",
+        "FromPort": 1858,
+        "ToPort": 1858,
+        "SourceSecurityGroupId": {
+          "Ref": "PSMSSHSG"
+        },
+        "Description": "Vault to PSMSSH connection"
+      }
+    },
+    "VaultSGIngress7": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {
+          "Ref": "VaultSG"
+        },
+        "IpProtocol": "icmp",
+        "FromPort": -1,
+        "ToPort": -1,
+        "SourceSecurityGroupId": {
+          "Ref": "VaultSG"
+        },
+        "Description": "Vault to Vault icmp connection"
```

Comment:
better be ICMP, capital...
Vault to Vault? I would change that to just "Vault ICMP connection"